modules/tinymce/src/core/main/ts/keyboard/SpaceKey.ts

```diff
@@ -8,14 +8,17 @@
 import Editor from '../api/Editor';
 import { EditorEvent } from '../api/util/EventDispatcher';
 import VK from '../api/util/VK';
+import { fireFakeInputEvent } from './FakeInputEvents';
 import * as InsertSpace from './InsertSpace';
 import * as MatchKeys from './MatchKeys';
 
 const executeKeydownOverride = (editor: Editor, evt: KeyboardEvent) => {
   MatchKeys.execute([
     { keyCode: VK.SPACEBAR, action: MatchKeys.action(InsertSpace.insertSpaceOrNbspAtSelection, editor) }
-  ], evt).each((_) => {
+  ], evt).each(() => {
     evt.preventDefault();
+    // Browsers sends space in data even if the dom ends up with a nbsp so we should always be sending a space
+    fireFakeInputEvent(editor, 'insertText', { data: ' ' });
```

Comment:
Just sanity checking - in DeleteBackspaceKeys we fire the fake event then preventDefault, and here we do the opposite order. That's intended/won't matter?

Reply:
It doesn't really matter but I tend to want the preventDefault early since if anything goes wrong like it throws an exception or something then the event would still be prevented. I will change that.